src/catalog/src/durable/persist.rs

```diff
@@ -922,22 +934,14 @@ impl PersistCatalogState {
         snapshot
     }
 
+    /// Listen and apply all updates currently in the catalog.
+    ///
+    /// Returns an error if this instance has been fenced out.
     #[mz_ore::instrument(level = "debug")]
     async fn sync_to_current_upper(&mut self) -> Result<(), CatalogError> {
         let upper = self.current_upper().await;
-        if upper != self.upper {
-            if self.is_read_only() {
-                self.sync(upper).await?;
-            } else {
-                // non-read-only catalogs do not know how to deal with other writers.
-                return Err(DurableCatalogError::Fence(format!(
-                    "current catalog upper {:?} fenced by new catalog upper {:?}",
-                    self.upper, upper
-                ))
-                .into());
-            }
-        }
-
+        self.sync(upper).await?;
+        self.epoch.validate()?;
```

Comment:
Looks like previously we would sync even if there was an upper mismatch, but if we were in a read_only mode (e.g. catalog-debug?). Did we lose that behavior?

Reply:
I don't think I fully understand the question. Now we always `sync`, even if there is an upper mismatch. This includes read_only mode.